packages/server/src/core/internals/getParseFn.ts

```diff
@@ -30,6 +30,17 @@ export function getParseFn<TType>(procedureParser: Parser): ParseFn<TType> {
     return parser.create.bind(parser);
   }
 
+  if (typeof parser.decode === 'function') {
+    // ProcedureParserIoTsEsque
+    return (value) => {
+      const decoded = parser.decode.bind(parser)(value);
+      if (decoded.left) {
+        throw decoded.left;
```

Comment:
Is the `.left` an `Error`? Make sure that we only `throw` Error objects. 

Ideally something that works together with error formatters! https://trpc.io/docs/error-formatting

Reply:
Yeah, that's exactly what I was talking about in my above thread. We should throw an Error when the Either is left.

Reply:
If there's a built-in error instance that can contain more metadata (similar to `ZodError`) that'd be perfect, otherwise we'd need to wrap it in our own custom wrapper

Reply:
On it

Reply:
There's a custom type yes.

There's also a library called io-ts-reporters that might come handy to get formated readable error messages for that matter. Not sure if it helps in the specific case.